File: SWEA/D3/10726. 이진수 표현/이진수 표현.py

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    n, m = map(int, input().split())

    bmarr = []

    result = "ON"
    for _ in range(n):
        if m % 2 == 0:
            result = "OFF"
            break
        m = m // 2

    print(f"#{test_case} {result}")

File: SWEA/D3/10726. 이진수 표현/이진수 표현.java
import java.io.FileInputStream;
import java.util.Scanner;

class Solution {
  public static void main(String args[]) throws Exception {

    Scanner sc = new Scanner(System.in);
    int T;
    T = sc.nextInt();

    for (int test_case = 1; test_case <= T; test_case++) {
      int n = sc.nextInt();
      int m = sc.nextInt();

      String result = "ON";
      for (int i = 0; i < n; i++) {
        int bit = 1 << i; // 비트 연산으로 0100 왼쪽으로 하나씩 미룬다
        if ((bit & m) != bit) { // & 연산으로 통과했을 때 bit과 같으면 그 자리가 1로 켜져있다.
          result = "OFF";
        }
      }

      System.out.println("#" + test_case + " " + result);
    }
  }
}
